tests/unit/views/TractionDashboard.spec.js

```diff
@@ -1,180 +1,181 @@
 import TractionDashboard from '@/views/TractionDashboard'
+import PipelinesConfig from '@/config/PipelinesConfig'
 import { mount, flushPromises } from '@support/testHelper'
 
 describe('TractionDashboard.vue', () => {
   let wrapper, box, dashboard
+  const active_pipelines = PipelinesConfig.filter((pipeline) => pipeline.active)
+  const active_pipeline_names = active_pipelines.map((pipeline) => pipeline.name)
 
   beforeEach(() => {
     wrapper = mount(TractionDashboard)
     dashboard = wrapper.vm
   })
 
   describe('pipelines', () => {
-    it('will have pipelines config', () => {
-      const config = dashboard.pipelines
-      expect(config.length).toEqual(3)
+    it('will have same number of active pipelines in config', () => {
+      const dashborad_pipelines = dashboard.pipelines
+      expect(dashborad_pipelines.length).toEqual(active_pipelines.length)
     })
 
-    it('will have saphyr config', () => {
-      const config = dashboard.pipelines
-      expect(config[0]['name']).toEqual('saphyr')
-      expect(config[0]['title']).toContain('Saphyr')
-    })
-
-    it('will have pacbio config', () => {
-      const config = dashboard.pipelines
-      expect(config[1]['name']).toEqual('pacbio')
-      expect(config[1]['title']).toEqual('PacBio')
-    })
+    it('will have the same active pipeline name in config', () => {
+      const dashborad_pipelines = dashboard.pipelines
+      const dashboard_pipeline_names = dashborad_pipelines.map((item) => item.name)
 
-    it('will have ont config', () => {
-      const config = dashboard.pipelines
-      expect(config[2]['name']).toEqual('ont')
-      expect(config[2]['title']).toEqual('ONT')
+      expect(dashboard_pipeline_names.length === active_pipeline_names.length)
+      expect(
+        dashboard_pipeline_names.every((value, index) => value === active_pipeline_names[index]),
+      )
     })
   })
 
-  describe('for saphyr', () => {
-    beforeEach(() => {
-      box = wrapper.find('[data-pipeline=Saphyr]')
-    })
-
-    it('will have a title', () => {
-      expect(box.find('[data-attribute=title]').text()).toEqual('Saphyr')
-    })
-
-    it('will have a description', () => {
-      expect(box.find('[data-attribute=description]').text()).toBeDefined()
-    })
-
-    describe('route buttons', () => {
-      it('will have a samples button', async () => {
-        const button = box.findAll('a')[0]
-        expect(button.text()).toEqual('Samples')
-        button.trigger('click')
-        await flushPromises()
-        expect(wrapper.vm.$route.path).toBe('/saphyr/samples')
+  if (active_pipeline_names.includes('saphyr')) {
```

Comment:
Conditionally running tests may not be ideal as it can cause non-deterministic tests and can be confusing. It would be better to use a mocking setup, which allows you to mock the file contents so we can control the testing data.

Reply:
Hi Seena, 
Thanks for your comments. I made a mock config file for testing(I push to git already) but I found it still not able to pass the test as the dashboard pipelines are read from real config file. so comparing dashboard pipeline and mock pipeline config file won't work. Any possible that the dashboard pipeline are from mock config file when testing?

Reply:
Hi Wendy, I think we don’t need to create a mock file instead we can mock what the file content is, for example , some thing like this
vi.mock(‘@/config/PipelinesConfig’, () => (
{
    “name”: “pipeline”,
    “title”: “Pipileline title”,
    “description”: “Testing”,
    “routes”: [“samples”, “libraries”, “runs”],
     active : false
  },
 ))

I am happy to catchup tomorrow, possibly in the afternoon, if you would like to discuss further

Reply:
Hi Seena,
Thanks for your suggestion. I made further changes for the test. Please review again.